go/cmd/dolt/commands/diff.go

```diff
@@ -816,6 +962,37 @@ func diffUserTables(queryist cli.Queryist, sqlCtx *sql.Context, dArgs *diffArgs)
 			continue
 		}
 
+		// Apply table-level filtering based on diff type
+		if dArgs.filter != nil && dArgs.filter.filterBy != NoFilter {
+			shouldIncludeTable := false
+
+			// Check if table was added
+			if delta.IsAdd() && dArgs.filter.includeAddsOrAll() {
+				shouldIncludeTable = true
+			}
+
+			// Check if table was dropped
+			if delta.IsDrop() && dArgs.filter.includeDropsOrAll() {
+				shouldIncludeTable = true
+			}
+
+			// Check if table was modified (schema change or rename)
+			if !delta.IsAdd() && !delta.IsDrop() {
+				isModified := delta.SchemaChange || delta.IsRename()
+				if isModified && dArgs.filter.includeModificationsOrAll() {
+					shouldIncludeTable = true
+				}
+				// If no schema/rename changes but has data changes, let it through for row-level filtering
+				if !isModified && delta.DataChange {
+					shouldIncludeTable = true
+				}
+			}
+
+			if !shouldIncludeTable {
+				continue // Skip this table but continue processing others
+			}
+		}
+
```

Comment:
Use the `TableDeltaSummary` object provided by `GetSummary()` member func in `table_deltas.go`. This may require rewriting your impl for the struct filter struct above, it also keeps the logic in one place. There should be a `DiffType` field you can make use of.